### PR TITLE
Added doc strings for swagger documentation

### PR DESCRIPTION
**Pull request by lukas-bednar**
Added documentation and align with Kubernetes naming for KubeVirt API calls.

### CODE REVIEW

**Comment by lukas-bednar** [comment]
Here is generated pysdk with documentation strings visible.
https://github.com/lukas-bednar/kubevirt-pysdk-generated/blob/kubevirt-master/docs/ApiskubevirtiovalphaApi.md

File: pkg/virt-api/rest/kubeproxy.go [hunk 1]
```diff
@@ -48,14 +48,15 @@ type ResponseHandlerFunc func(rest.Result) (interface{}, error)
...
+	ws.Doc("The KubeVirt API, a virtual machine management.")
 	ws.Path(GroupVersionBasePath(gv))
 
 	virtClient, err := kubecli.GetKubevirtClient()
 	if err != nil {
 		return nil, err
 	}
 	autodiscover := endpoints.NewHandlerBuilder().Get().Decoder(endpoints.NoopDecoder).Endpoint(NewAutodiscoveryEndpoint(virtClient.RestClient())).Build(ctx)
-	ws.Route(ws.GET("/").Produces(mime.MIME_JSON).Writes(metav1.APIResourceList{}).To(endpoints.MakeGoRestfulWrapper(autodiscover)))
+	ws.Route(ws.GET("/").Produces(mime.MIME_JSON).Writes(metav1.APIResourceList{}).To(endpoints.MakeGoRestfulWrapper(autodiscover)).Doc("Get KubeVirt API Resources"))
```
Comment (lukas-bednar) [inline pkg/virt-api/rest/kubeproxy.go]: I would like to set `Operation` attribute. do you have any preference?
`list_resources` / `get_resources`

**Comment by rmohr** [comment]
@lukas-bednar works for me in general. I think I was a little bit too fast with adding the Operation names. It would be great if we could align our names with the ones of `apis` and `pod` with Kubernetes. Could you also cross-check the `Operation` fields which I have set with the ones from k8s too?

Interesting links:
 * `apis`: https://github.com/kubernetes/kubernetes/blob/master/api/swagger-spec/apis.json
 * example for all namespaces naming: https://github.com/kubernetes/kubernetes/blob/master/api/swagger-spec/v1.json#L9861

**Comment by lukas-bednar** [comment]
@rmohr Thanks for hints, I took a look at that and I will change `Operation` for namespaced calls like:

**POST** -> `"createNamespaced" + objKind`
**PUT** -> `"replaceNamespaced" + objKind`
**PATCH** -> `"updateNamespaced" + objKind`
**DELETE** -> `"deleteNamespaced" + objKind`
**GET** -> `"{read , watch }Namespaced" + objKind`
**GET** -> `"listNamespaced" + objKind + "List"`

and for all namespaces:

**GET** -> `"{read , watch }" + objKind + "ForAllNamespaces"`
**GET** -> `"listNamespaced" + objKind + "ListForAllNamespaces"`

in additional I am going to add `swagger.Info{}` into `swagger.Config{}` since swagger-generator complains about absence ... I am not sure about some items ...
```go
		Info:            swagger.Info{
			Title:             "KubeVirt API, ",
			Description:       "This is KubeVirt API an add-on for Kubernetes.",
			TermsOfServiceUrl: "???",  // I don't know what to put there
			Contact:           "kubevirt-dev@googlegroups.com",
			License:           "Apache 2.0",
			LicenseUrl:        "https://www.apache.org/licenses/LICENSE-2.0",
		},
```

Regarding to `apis` I am not sure what should I do there.
What I am missing there (when I compare it to kubernetes), are these API groups and versions.
Do you want me to add `APIGroupList` under `"/kubevirt.io/"`, to serve list of available API versions ?

File: cmd/virt-api/virt-api.go [hunk 2]
```diff
@@ -127,6 +127,14 @@ func (app *virtAPIApp) Run() {
 		ApiPath:         "/swaggerapi",
 		SwaggerPath:     "/swagger-ui/",
 		SwaggerFilePath: app.SwaggerUI,
+		Info: swagger.Info{
```
Comment (lukas-bednar) [inline cmd/virt-api/virt-api.go]: Even having this here, it doesn't appear in final json file: 
```json
{
  "swaggerVersion": "1.2",
  "apiVersion": "",
  "basePath": "http://localhost:8183",
  "resourcePath": "/apis/kubevirt.io/v1alpha1",
  "info": {
   "title": "",
   "description": ""
  },
```

**Comment by lukas-bednar** [comment]
Here you can see generated PySDK https://github.com/lukas-bednar/kubevirt-pysdk-generated/blob/master/docs/ApiskubevirtIov1alpha1Api.md#console

@fabiand  I also generated GitBook for this revision, and it doesn't look good. I will try to understand what is problem.

**Comment by lukas-bednar** [comment]
@fabiand I found it, that plugin doesn't support specification 1.2, when converting it to 2.0, it started rendering without problem: https://www.gitbook.com/book/lukas-bednar/kubevirt-api-doc/details

File: pkg/virt-api/rest/kubeproxy.go [hunk 3]
```diff
@@ -87,76 +90,86 @@ func GenericResourceProxy(ws *restful.WebService, ctx context.Context, gvr schem
...
 	))
 
 	ws.Route(addGetAllNamespacesListParams(
 		ws.GET(gvr.Resource).
 			Produces(mime.MIME_JSON, mime.MIME_YAML).
-			Operation("list_all_"+plural).
-			To(endpoints.MakeGoRestfulWrapper(getListAllNamespaces)).Writes(listExample).Doc("test4"), ws,
+			Operation("list"+objKind+"ForAllNamespaces").
+			To(endpoints.MakeGoRestfulWrapper(getListAllNamespaces)).Writes(listExample).
+			Doc("Get a list all of "+objKind+" objects."), ws,
```
Comment (rmohr) [inline pkg/virt-api/rest/kubeproxy.go]: Get a list *of* all

File: cmd/virt-api/virt-api.go (hunk 2 above)
Comment (rmohr) [reply cmd/virt-api/virt-api.go]: It is there,  when you look at `/swaggerapi` (the base path). You won't see it under `/swaggerapi/apis/kubevirt.io/v1alpha1`. I hope that this is good enough, if you use the /swaggerapi endpoint as the entrypoint for swagger-gen.

File: cmd/virt-api/virt-api.go [hunk 4]
```diff
@@ -127,6 +127,14 @@ func (app *virtAPIApp) Run() {
...
+			Title:       "KubeVirt API, ",
+			Description: "This is KubeVirt API an add-on for Kubernetes.",
+			//			TermsOfServiceUrl: "???",  // I don't know what to put there
```
Comment (rmohr) [inline cmd/virt-api/virt-api.go]: If you e.g. host a service, and you need to distribute usage-terms, this is the  place where you can specify a URL. I don't think it is important for us.

Comment (lukas-bednar) [reply cmd/virt-api/virt-api.go]: OK U will remove it then

File: pkg/virt-api/rest/kubeproxy.go (hunk 3 above)
Comment (lukas-bednar) [reply pkg/virt-api/rest/kubeproxy.go]: Done

**Comment by fabiand** [comment]
@lukas-bednar how did you perform the conversion from 1.2 to 2.0 as you mentioned above.

All in all ti looks nice, especially the result:
https://lukas-bednar.gitbooks.io/kubevirt-api-doc/content/

**Comment by lukas-bednar** [comment]
> @lukas-bednar how did you perform the conversion from 1.2 to 2.0 as you mentioned above.

@fabiand There is [api-spec-converter](https://github.com/LucyBot-Inc/api-spec-converter) .

**Pull request merged**